tests/app/forms/validation/test_number_validator.py

```diff
@@ -26,6 +23,20 @@ def test_number_validator_raises_StopValidation(
     assert error_messages["INVALID_NUMBER"] == str(exc.value)
 
 
+@pytest.mark.usefixtures("gb_locale")
+def test_number_validator_NaN_raises_StopValidation(
+    number_check, mock_form, mock_field
+):
+    value = "NaN"
```

Comment:
Can you not just do this in the one above?. Just always set the data given that is the real scenario. Then you don't need a dedicated test here.

Reply:
Yeah but in above scenario Decimal will be casted on "a"  and try except will stop on exception but you cannot cast decimal on "a" in preparation before you run number check, but at the point where isnan checked Decimal is casted on NaN already. So I had to  created this separate. Basically you cannot do
```python
mock_field.data = Decimal(value)
```
 in:
```python
@pytest.mark.parametrize(
    "value",
    ([None], [""], ["a"], ["2E2"]),
)
@pytest.mark.usefixtures("gb_locale")
def test_number_validator_raises_StopValidation(
    number_check, value, mock_form, mock_field
):
    mock_field.raw_data = value
    mock_field.data = Decimal(value)

    with pytest.raises(StopValidation) as exc:
        number_check(mock_form, mock_field)

    assert error_messages["INVALID_NUMBER"] == str(exc.value)
```
when the value is "a"

Reply:
Fair enough. 👍